src/Project.ts

```diff
@@ -146,32 +168,34 @@ export default class Project {
     );
   }
 
-  render() {
+  private render(): void {
     // Render to canvas
-    this.renderer.update(this.stage, this.spritesAndClones);
+    this.renderer.update();
 
     // Update watchers
-    for (const sprite of [...Object.values(this.sprites), this.stage]) {
-      for (const watcher of Object.values(sprite.watchers)) {
-        watcher.updateDOM(this.renderer.renderTarget);
+    if (this.renderer.renderTarget) {
+      for (const sprite of [...Object.values(this.sprites), this.stage]) {
+        for (const watcher of Object.values(sprite!.watchers)) {
+          watcher!.updateDOM(this.renderer.renderTarget);
+        }
       }
     }
   }
 
-  _renderLoop() {
+  private _renderLoop(): void {
     requestAnimationFrame(this._renderLoop.bind(this));
     this.render();
   }
 
-  fireTrigger(trigger, options) {
+  public fireTrigger(trigger: symbol, options?: TriggerOptions): Promise<void> {
     // Special trigger behaviors
     if (trigger === Trigger.GREEN_FLAG) {
       this.restartTimer();
       this.stopAllSounds();
       this.runningTriggers = [];
 
       for (const spriteName in this.sprites) {
-        const sprite = this.sprites[spriteName];
+        const sprite = this.sprites[spriteName]!;
```

Comment:
I'm pretty sure the `!` is only necessary here since `sprites` is a `Partial`. If `sprites` is rewritten without partial, this is probably safe to remove.

Reply:
As discussed we're using `Partial<Record>` which justifies the need for a "this definitely isn't null" assertion here.